platform/jewel/foundation/src/main/kotlin/org/jetbrains/jewel/foundation/lazy/SelectableLazyColumn.kt

```diff
@@ -229,9 +236,30 @@ private fun Modifier.selectable(
         }
         // Add semantics for accessibility
         .semantics(mergeDescendants = true) {
-            selected = itemKey in selectableState.selectedKeys
+            val isSelected = itemKey in selectableState.selectedKeys
+
+            selected = isSelected
+            toggleableState = ToggleableState(isSelected)
+
             focused = selectableState.lastActiveItemIndex == allKeys.indexOfFirst { it.key == itemKey }
-            stateDescription = ""
+            isTraversalGroup = false
+
+            role =
+                when (selectionMode) {
+                    SelectionMode.Multiple -> Role.Checkbox
+                    else -> Role.RadioButton
+                }
+
+            onClick {
+                actionHandler.toggleKeySelection(
+                    selectableLazyListState = selectableState,
+                    selectionMode = selectionMode,
+                    allKeys = allKeys,
+                    key = itemKey,
+                )
+                requester?.requestFocus()
```

Comment:
Calling requestFocus before was causing some intermitent automatically selection of the first item based in the focus logic:

```kt
.onFocusChanged {
    isFocused = it.hasFocus
    with(state) {
        if (isFocused && lastActiveItemIndex == null && selectedKeys.isEmpty()) {
            keyActions.actions.onSelectFirstItem(keys, this)
        }
    }
}
```